include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -96,4 +97,56 @@ def XTenNN_OutputOp : XTenNN_Op<"output", [
 }
 
 
+def XTenNN_QuantizeOp: XTenNN_Op<"quantize", [
+    SameOperandsAndResultShape]> {
+  
+
+  let summary = "quantizes a float32 tensor to a signed or unsigned integer tensor of given width.";
+
+  let description = [{
+    Quantizes a given float32 tensor into a signed or unsigned integer tensor of given width.
+
+    Applies the following to linear quantization the input tensor x:
+      y = round( x * scale ) + zero_point
+
+      Where scale is an attribute of the operation in f32 and is a power-of-two value.
+      While the zero_point is always assumed to be zero.
+  }];
+
+  let arguments = (ins 
+    F32Tensor:$input,
```

Comment:
What about F64, F16, BFloat16 inputs? Is there a reason not to allow any floating point input?